amplfi/train/models/flow.py

```diff
@@ -124,6 +126,66 @@ def on_test_epoch_start(self):
         self.test_results: list[Result] = []
         self.idx = 0
 
+    def _get_log_prior_dict(self, trainer=None):
+        """
+        Get the log prior for each parameter in the model.
+        """
+        if trainer is not None:
+            self.trainer = trainer
+        log_prior_dict = {}
+        sf = self.trainer.datamodule.waveform_sampler.parameter_sampler # need this to get the prior for some parameters
+
+        for param in self.inference_params:
+            if param in sf.parameters:
+                # some priors live in the ```waveform_sampler.parameter_sampler``` object
+                # {chirp_mass': Uniform(low: 10.0, high: 100.0),
+                # 'mass_ratio': Uniform(low: 0.125, high: 0.9990000128746033),
+                # 'distance': PowerLaw(),
+                # 'inclination': Sine(),
+                # 'phic': Uniform(low: 0.0, high: 6.2831854820251465),
+                # 'chi1': DeltaFunction(),
+                # 'chi2': DeltaFunction()}
+                log_prior_dict[param] = sf.parameters[param]
+            else:
+                # others are seperately
+                # waveform_sampler."name"
+                # 'dec': Cosine(),
+                # 'psi': Uniform(low: 0.0, high: 3.140000104904175),
+                # 'phi': Uniform(low: -3.140000104904175, high: 3.140000104904175)
+                log_prior_dict[param] = getattr(self.trainer.datamodule.waveform_sampler, param)
+        # add low and high boundaries (for some parameters these values are missing)
+        log_prior_dict['distance'].low = 100
+        log_prior_dict['distance'].high = 3100
+        log_prior_dict['inclination'].low = 0
+        log_prior_dict['inclination'].high = np.pi
+        log_prior_dict['dec'].low = -np.pi/2
+        log_prior_dict['dec'].high = np.pi/2
+        return log_prior_dict
+
+    def filter_descaled_parameters(self, descaled):
+        """
+        Filter the descaled parameters to keep only valid samples within their boundaries.
+        
+        Args:
+            descaled (torch.Tensor): The descaled parameters tensor.
+        Returns:
+            torch.Tensor: The filtered descaled parameters.
+        """
+        valid_idxs = torch.ones(descaled.shape[0], dtype=torch.bool, device=descaled.device)
+        num_discarded = 0
+        for idx, param in enumerate(self.inference_params):
+            prior = self.log_prior_dict[param]
+            low = prior.low
+            high = prior.high
+            valid_idxs &= (descaled[:, idx] >= low) & (descaled[:, idx] <= high)
+            discarded_count = (~valid_idxs).sum().item()
+            num_discarded += discarded_count
+            print(f"Discarded samples[{param}]: {discarded_count}")
```

Comment:
Instead of print, you should be able to use `self._logger.info` to use the model logger object